enterprise/backend/src/metabase_enterprise/sandbox/models/params/field_values.clj

```diff
@@ -75,6 +76,26 @@
                                          (mbql.u/match-one v [:dimension [:field field-id _]] field-id))]
                     {k (get login-attributes k)})))])))
 
+(defenterprise field-id->field-values-for-current-user
+  "Fetch *existing* FieldValues for a sequence of `field-ids` for the current User. Values are returned as a map of
```

Comment:
Dan: 

Can you clarify the expectation here? The docstring stresses this fetches *existing* FieldValues. It seems the unsandboxed fields behave this way (params.field-values/default-field-id->field-values-for-current-user only selects from the app db), but the sandboxed version calls a get-or-create style function which seems to create FieldValues that do not yet exist.

I guess I'm confused why existence is a relevant factor for return value. To me, it seems like the existence of FieldValues is a bit of an implementation detail. My thinking is both sandboxed and not should create field values if they don't exist. Can you explain the thinking behind the divergence here?

https://github.com/metabase/metabase/pull/29359#discussion_r1170002315

Reply:
there are 2 main types of FieldValues: full and advanced.

In this case sandboxed field values are considered `advanced` FieldValues and we consider them as "on-demand" field values. 

Unlike the full field values, we scan them and store them during sync and do not think of them as `on-demand`.

Reply:
that explains why we did not have `get-or-create` for Full FieldValues.

But, since we introduced [Disable field value scanning for inactive fields](https://github.com/metabase/metabase/pull/26974) a while ago, I think it's also good to call `get-or-create` when fetching Full FieldValues too since it's now semi-on-demand.